test/src/test/java/io/pravega/schemaregistry/integrationtest/TestEndToEndWithAuth.java

```diff
@@ -0,0 +1,111 @@
+import io.pravega.schemaregistry.client.SchemaRegistryClient;
+import io.pravega.schemaregistry.client.SchemaRegistryClientConfig;
+import io.pravega.schemaregistry.client.SchemaRegistryClientFactory;
+import io.pravega.schemaregistry.common.CredentialProvider;
+import io.pravega.schemaregistry.server.rest.RestServer;
+import io.pravega.schemaregistry.server.rest.ServiceConfig;
+import io.pravega.schemaregistry.service.Config;
+import io.pravega.schemaregistry.service.SchemaRegistryService;
+import io.pravega.schemaregistry.storage.SchemaStore;
+import io.pravega.test.common.TestUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.curator.shaded.com.google.common.base.Charsets;
+import org.junit.After;
+import org.junit.Before;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URI;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Base64;
+import java.util.concurrent.Executors;
+
+@Slf4j
+public class TestEndToEndWithAuth extends TestInMemoryEndToEnd {
+    private final static String SYSTEM_ADMIN = "admin";
+    private final static String PASSWORD = "1111_aaaa";
+
+    protected int port;
+    private RestServer restServer;
+    private File authFile;
+
+    @Before
+    @Override
+    public void setUp() {
+        executor = Executors.newScheduledThreadPool(Config.THREAD_POOL_SIZE);
+
+        port = TestUtils.getAvailableListenPort();
+        authFile = createAuthFile();
+        ServiceConfig serviceConfig = ServiceConfig.builder().port(port)
+                                                   .authEnabled(true)
+                                                   .userPasswordFilePath(authFile.getAbsolutePath())
+                                                   .build();
+        SchemaStore store = getStore();
+
+        SchemaRegistryService service = new SchemaRegistryService(store, executor);
+
+        restServer = new RestServer(service, serviceConfig);
+        restServer.startAsync();
+        restServer.awaitRunning();
+    }
+
+    @After
+    public void tearDown() {
+        restServer.stopAsync();
+        restServer.awaitTerminated();
+        executor.shutdownNow();
+        authFile.delete();
+    }
+
+    @Override
+    public SchemaRegistryClient getClient() {
+        return SchemaRegistryClientFactory.withDefaultNamespace(
+                SchemaRegistryClientConfig.builder().schemaRegistryUri(URI.create("http://localhost:" + port))
+                                          .authentication(new CredentialProvider.DefaultCredentialProvider("Basic", 
+                                                  Base64.getEncoder().encodeToString((SYSTEM_ADMIN + ":" + PASSWORD).getBytes(Charsets.UTF_8))))
+                                          .build());
+    }
+    
+    private File createAuthFile() {
+        try {
+            File authFile = File.createTempFile("auth_file", ".txt");
```

Comment:
`io.pravega.test.integration.utils.PasswordAuthHandlerInput` in Pravega is a helper class for creating and populating input files for the PasswordAuthHandler. You might want to use that here.

Reply:
we dont have dependency on pravega test module available. so dont have access to that utility class.